src/spec/create_extension_spec.py

```diff
@@ -59,11 +60,49 @@ def main():
         name='session_performance',
         doc='Performance of correct responses in %',
         dtype='float',
+        required=False
+    )
+
+    RigExtension = NWBGroupSpec(
+        doc='type for storing rig information',
+        neurodata_type_def='RigExtension',
+        neurodata_type_inc='LabMetaData',
+    )
+
+    rig_attr = ['rig', 'simulationMode', 'hasDAQ', 'hasSyncComm', 'minIterationDT', 'arduinoPort',
+                'sensorDotsPerRev', 'ballCircumference', 'toroidXFormP1', 'toroidXFormP2',
+                'colorAdjustment', 'soundAdjustment', 'nidaqDevice', 'nidaqPort', 'nidaqLines',
+                'syncClockChannel', 'syncDataChannel', 'rewardChannel', 'rewardSize',
+                'rewardDuration', 'laserChannel', 'rightPuffChannel', 'leftPuffChannel',
+                'webcam_name']
+    for attr in rig_attr:
+        RigExtension.add_attribute(
+            name=attr,
+            doc='rig information',
+            dtype='text',
+        )
+
+    LabMetaDataExtension.add_group(
+        name='rig',
+        neurodata_type_inc='RigExtension',
+        doc='type for storing rig information',
+    )
+
+    MazeExtension = NWBGroupSpec(
+        doc='type for storing maze information',
+        neurodata_type_def='MazeExtension',
+        neurodata_type_inc='DynamicTable',
```

Comment:
Why a dynamic table?

Reply:
Because it can have several rows, depends on each file how many.